### PR TITLE
support for views component persistence

### PR DESCRIPTION
**Pull request by sns-seb**
this PR contains changes in CE to implement Component persistence in Views

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/input' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/input) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/output' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/output) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/db/protobuf' package (org.codehaus.sonar:sonar-db:src/main/gen-java/org/sonar/db/protobuf) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonarqube/ws' package (org.codehaus.sonar:sonar-ws:src/main/gen-java/org/sonarqube/ws) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java [hunk 1]
```diff
@@ -20,282 +20,367 @@
...
+    componentDto.setDeprecatedKey(componentKey);
+    componentDto.setEnabled(true);
+    componentDto.setCreatedAt(new Date(system2.now()));
+    return componentDto;
+  }
+
+  /**
+   * Applies to a node of type either MODULE, SUBVIEW, PROJECT_VIEW
+   */
+  private void setRootAndParentModule(ComponentDto res, PathAwareVisitor.Path<ComponentDtoHolder> path) {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java]: should be static

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java [hunk 2]
```diff
@@ -58,6 +57,9 @@
 import org.sonar.server.util.RubyUtils;
 
 import static com.google.common.collect.Lists.newArrayList;
+import static org.sonar.db.component.ComponentDtoFunctions.toCopyResourceId;
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java]: These modifications would deserve to be in another commit

File: server/sonar-server/src/main/java/org/sonar/server/test/ws/CoveredFilesAction.java [hunk 3]
```diff
@@ -104,7 +105,7 @@ private static void writeTests(List<CoveredFileDoc> coveredFiles, Map<String, Co
     } finally {
       MyBatis.closeQuietly(dbSession);
     }
-    return Maps.uniqueIndex(components, new ComponentToUuidFunction());
+    return Maps.uniqueIndex(components, ComponentDtoFunctions.toUuid());
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/test/ws/CoveredFilesAction.java]: Should have been in another commit

File: server/sonar-server/src/main/java/org/sonar/server/test/ws/ListAction.java [hunk 4]
```diff
@@ -169,7 +170,7 @@ private static long coveredLines(List<CoveredFileDoc> coveredFiles) {
     List<String> fileUuids = Lists.transform(tests, new TestToFileUuidFunction());
     List<ComponentDto> components = dbClient.componentDao().selectByUuids(dbSession, fileUuids);
 
-    return Maps.uniqueIndex(components, new ComponentToUuidFunction());
+    return Maps.uniqueIndex(components, ComponentDtoFunctions.toUuid());
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/test/ws/ListAction.java]: Should have been done in another commit

**Comment by julienlancelot** [comment]
Minor feedbacks and LGTM

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java]: :+1:

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java (hunk 2 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java]: they are

File: server/sonar-server/src/main/java/org/sonar/server/test/ws/CoveredFilesAction.java (hunk 3 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/test/ws/CoveredFilesAction.java]: it is

File: server/sonar-server/src/main/java/org/sonar/server/test/ws/ListAction.java (hunk 4 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/test/ws/ListAction.java]: it is

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/input' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/input) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/output' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/output) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/db/protobuf' package (org.codehaus.sonar:sonar-db:src/main/gen-java/org/sonar/db/protobuf) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonarqube/ws' package (org.codehaus.sonar:sonar-ws:src/main/gen-java/org/sonarqube/ws) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
